biocypher/_get.py

```diff
@@ -80,74 +101,79 @@ def __init__(self, cache_dir: Optional[str] = None) -> None:
         self.cache_dict = self._load_cache_dict()
 
     # download function that accepts a resource or a list of resources
-    def download(self, *resources: Resource):
+    def download(self, *downloads: Union[Resource, APIRequest]):
         """
-        Download one or multiple resources.
+        Download one or multiple resources, APIrequest, or both.
 
         Args:
-            resources (Resource): The resource or resources to download.
+            downloads (Resource or APIRequest): The resource(s) or API request(s) to download.
 
         Returns:
-            str or list: The path or paths to the downloaded resource(s).
+            list[str]: The path or paths to the downloaded resource(s) or API request(s).
         """
         paths = []
-        for resource in resources:
-            paths.append(self._download_or_cache(resource))
+        for download in downloads:
+            paths.append(self._download_or_cache(download))
 
         # flatten list if it is nested
         if is_nested(paths):
             paths = [path for sublist in paths for path in sublist]
 
         return paths
 
-    def _download_or_cache(self, resource: Resource, cache: bool = True):
+    def _download_or_cache(
+        self, download: Union[Resource, APIRequest], cache: bool = True
+    ):
         """
-        Download a resource if it is not cached or exceeded its lifetime.
+        Download a resource or an API request if it is not cached or exceeded its lifetime.
 
         Args:
-            resource (Resource): The resource to download.
-
+            download (Resource or APIRequest): The resource or API request to download.
         Returns:
-            str or list: The path or paths to the downloaded resource(s).
+            list[str]: The path or paths to the downloaded resource(s) or API request(s).
         """
-        expired = self._is_cache_expired(resource)
+        expired = self._is_cache_expired(download)
 
         if expired or not cache:
-            self._delete_expired_resource_cache(resource)
-            logger.info(f"Asking for download of {resource.name}.")
-            paths = self._download_resource(cache, resource)
+            self._delete_expired_cache(download)
+            if isinstance(download, Resource):
+                logger.info(f"Asking for download of resource {download.name}.")
+                paths = self._download_resource(cache, download)
+            else:
```

Comment:
I would check here for the instance of the API class. Not safe to assume that everything that is not a `Resource` is an `APIRequest`. Better to test explicitly.